src/assets/Keycloak/ocis-with-keycloak.md

```diff
@@ -0,0 +1,161 @@
+      traefik.http.routers.ocis.rule: PathPrefix(`/`)
+      traefik.http.routers.ocis.entrypoints: ocis
+      traefik.http.services.ocis.loadbalancer.server.port: 9200
+
+  postgres:
+    image: postgres:alpine
+    volumes:
+      - postgres_data:/var/lib/postgresql/data
+    environment:
+      POSTGRES_DB: keycloak
+      POSTGRES_USER: keycloak
+      POSTGRES_PASSWORD: keycloak
+
+  keycloak:
+    image: quay.io/keycloak/keycloak:24.0.1
+    command: ["start-dev", "--proxy=edge"]
+    environment:
+      KC_DB: postgres
+      KC_DB_URL: "jdbc:postgresql://postgres:5432/keycloak"
+      KC_DB_USERNAME: keycloak
+      KC_DB_PASSWORD: keycloak
+      KC_FEATURES: impersonation
+      KEYCLOAK_ADMIN: admin
+      KEYCLOAK_ADMIN_PASSWORD: admin
+    extra_hosts:
+      - host.docker.internal:host-gateway
+    labels:
+      traefik.enable: true
+      traefik.http.routers.keycloak.tls: true
+      traefik.http.routers.keycloak.rule: PathPrefix(`/`)
+      traefik.http.routers.keycloak.entrypoints: keycloak
+      traefik.http.services.keycloak.loadbalancer.server.port: 8080
+    depends_on:
+      - postgres
+
+volumes:
+  postgres_data:
+```
+
+
+### Manual configuration for ocis on keycloak
+1. Go to ocis url: `https://host.docker.internal:9200/`
+![ocis](src/assets/keycloak/images/ocis-keycloak-UI-setup-1.png)
+
+2. Go to keycloak url: `https://host.docker.internal:8443/`
+![keycloak](src/assets/keycloak/images/ocis-keycloak-UI-setup-2.png)
+
+3. Create new realm: `ocis`
+![create realm](src/assets/keycloak/images/create-realm.png)
+![create realm with name `ocis`](src/assets/keycloak/images/create-realm-2.png)
+
+4. Add a new client:
+![client registration](src/assets/keycloak/images/client-registration.png)
+
+- Client ID: `web`
+![add client](src/assets/keycloak/images/client-registration-2.png)
+
+- Root URL: `https://host.docker.internal:9200`
+![add root url](src/assets/keycloak/images/client-registration-3.png)
+
+5. Add realm roles:
+
+- Role name: `ocisAdmin`
+![add realm role](src/assets/keycloak/images/create-realm-role.png)
+
+6. Create a new user:
+
+- Username: `admin` and other info
+![create user](src/assets/keycloak/images/create-user-1.png)
+![add user information](src/assets/keycloak/images/create-user-2.png)
+
+- Create password
+![create password](src/assets/keycloak/images/create-user-3.png)
+![setup password for user](src/assets/keycloak/images/create-user-4.png)
+
+- Role Mapping: Assign `ocisAdmin` role
+![assign role](src/assets/keycloak/images/assign-role-to-user.png)
+![assign role to user](src/assets/keycloak/images/asign-role-to-user-2.png)
+
+7. Update `roles` Client scope: `Client Scopes` -> `roles`
+- `Mappers` -> `realm roles` -> `Token Claim Name=roles`
+![update-roles](src/assets/keycloak/images/update-roles.png)
+
+8. Go to url `https://host.docker.internal:9200/` reload if already opened
+![login ocis with keycloak](src/assets/keycloak/images/ocis-keycloak-login.png)
```

Comment:
I'm at the end of the blog post, but what did I achieve?

Reply:
Short conclusion/summary would be nice. What is the end result after following these steps? What do I need to be aware of if I want to replicated them to my own project?